src/c++/perf_analyzer/genai-pa/genai_pa/llm_metrics.py

```diff
@@ -121,6 +121,65 @@ def __repr__(self):
         attr_strs = ",".join([f"{k}={v}" for k, v in self.__dict__.items()])
         return f"Statistics({attr_strs})"
 
+    def _is_seconds_field(self, field: str):
+        seconds_fields = [
+            "inter_token_latency",
+            "time_to_first_token",
+            "end_to_end_latency",
+        ]
+        return field in seconds_fields
+
+    def _use_exact_length(self, value: str):
+        exact_value_length = 17
+        formatted_value = f"{value:.{exact_value_length}f}"
+        if len(formatted_value) < exact_value_length:
+            return formatted_value.ljust(exact_value_length)
+        return formatted_value[:exact_value_length]
+
+    def pretty_print(self):
+        field_stats = {}
+        print("Output:")
+
+        for key, value in self.__dict__.items():
+            if key.startswith(
+                (
+                    "p25_",
+                    "p50_",
+                    "p75_",
+                    "p90_",
+                    "p95_",
+                    "p99_",
+                    "avg_",
+                    "min_",
+                    "max_",
+                    "std_",
+                )
+            ):
+                stat, field = key.split("_", 1)
+                stat = stat.replace("_", " ")
+
+                if field not in field_stats:
+                    field_stats[field] = {}
+                field_stats[field][stat] = value
+
+        for field, stats in field_stats.items():
+            is_seconds_field = self._is_seconds_field(field)
+
+            field_label = f"{field}"
+            if is_seconds_field:
+                field_label += "_s"
```

Comment:
1. What is the unit here? I assume s means seconds, but the values in the screenshot are huge for seconds.

2. We should probably truncate the number of digits after decimal place with `{:.2f}` or similar

Reply:
The units are in seconds. This ticket is only meant to be about formatting, so @debermudez is going to get back to me on the units used in the calculations.

Thanks for clarifying the units. It was mentioned at standup, but I forgot to mention this in the pull request to keep it self-documenting. I'll update the description.

For the second point, I'll defer to @debermudez. I tried to follow the given template. There was not a consistent length for the values, but they were about ~17 characters long. So for consistency, I made all values have a length of 17 by calling _use_exact_length(). We can use a smaller value or use a different formatting rule (e.g. two digits after the decimal place). I'm open.

Reply:
Do you think there'd be value in a "rich" table output, similar to some of my other CLIs (ex: glab_job_history.py)?

Seems like a good candidate for a condensed table view where rows are all these p25-max and columns are the ttft, itl, e2e metrics, rather than a long vertical view possibly requiring some scrolling

Reply:
I would be open. I'm a fan of the rich table output. It really depends on the decision makers in Tools. I don't know if that would take away from customers' ability to easily parse CLI data or would be incongruent with the other Triton tools, but it could be a nice user experience.

Reply:
Is there a sample of the rich table output I can view?
i dont know where glab_job_history.py to view it.

Reply:
> Is there a sample of the rich table output I can view?

Sure, here's one:

![image](https://github.com/triton-inference-server/client/assets/21284872/db6b0143-7dcc-41a2-962e-643a46244975)

Rich library: https://rich.readthedocs.io/en/stable/reference/table.html

Reply:
Sure, here's an example:

![image](https://github.com/triton-inference-server/client/assets/21284872/d69c5965-9c18-4acc-8aea-328a76663426)


Rich library: https://rich.readthedocs.io/en/stable/tables.html

Example code:
```python
from rich.console import Console
from rich.table import Table
import random

table = Table(title="PA LLM Metrics")

table.add_column("Statistic", justify="right", style="cyan", no_wrap=True)
table.add_column("Inter-token Latency (s)", justify="right", style="green")
table.add_column("Time to Tirst Token (s)", justify="right", style="green")

for stat in ["p25", "p50", "p75", "p90", "p95", "p99", "min", "mean", "max"]:
    itl = random.random()
    ttft = random.random()
    table.add_row(stat, f"{itl:.6f}", f"{ttft:.6f}")

console = Console()
console.print(table)
```

Reply:
did you really just whip up a solution? crazy

I love this. 

Is it possible to do 
max, mean, p99
line
p95 ->  p25 in descending order

Reply:
You can do pretty much whatever you want, it's a pretty flexible library. This was my interpretation of your ask, but you can modify and try yourself as much as you like:
![image](https://github.com/triton-inference-server/client/assets/21284872/fbf006e8-b9fd-48ec-a602-44846fa09583)

```python
from rich.console import Console
from rich.table import Table
import random

table = Table(title="PA LLM Metrics")

table.add_column("Statistic", justify="right", style="cyan", no_wrap=True)
table.add_column("Inter-token Latency (s)", justify="right", style="green")
table.add_column("Time to Tirst Token (s)", justify="right", style="green")

for stat in ["max", "mean", "p99", "LINE", "p95", "p90", "p75", "p50", "p25", "min"]:
    itl = random.random()
    ttft = random.random()
    if stat == "LINE":
        table.add_section()
    else:
        table.add_row(stat, f"{itl:.6f}", f"{ttft:.6f}")

console = Console()
console.print(table)
```

I imagine the final code would look a bit different.

Reply:
ridiculous

Reply:
and by that i mean i very much appreciate your help. thanks man

Reply:
@debermudez @dyastremsky the unit is in nanoseconds!

Reply:
Thanks, Hyunjae!